Operator fails after deploying Tenant with buckets to create during provisioning

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->
First I deploy the MinIO Operator via Helm (chart: https://github.com/minio/operator/tree/master/helm/operator) and I am accepting all the default values:
```
helm install minio-operator minio/operator -n minio
```

This results in 2 minio-operator pods running and they seem to be happy.

Next I attempt to deploy a Tenant using Helm (chart: https://github.com/minio/operator/tree/master/helm/tenant) and in this case since the default `values.yaml` is very hard-coded, I override more of those values. Most importantly for this issue, I am attempting to create 2 buckets during provisioning so I have the `buckets` array specified below.

```yaml
## MinIO Tenant Definition
tenants:
  # Tenant name
  - name: minio1
    ## Customize namespace for tenant deployment
    namespace: minio
    ## Specification for MinIO Pool(s) in this Tenant.
    pools:
      ## Servers specifies the number of MinIO Tenant Pods / Servers in this pool.
      ## For standalone mode, supply 1. For distributed mode, supply 4 or more.
      ## Note that the operator does not support upgrading from standalone to distributed mode.
      - servers: 4
        ## volumesPerServer specifies the number of volumes attached per MinIO Tenant Pod / Server.
        volumesPerServer: 4
        ## size specifies the capacity per volume
        size: 100Mi  # 2Gi
        ## storageClass specifies the storage class name to be used for this pool
        storageClassName: local-path
        ## Used to specify a toleration for a pod
        tolerations: { }
        ## nodeSelector parameters for MinIO Pods. It specifies a map of key-value pairs. For the pod to be
        ## eligible to run on a node, the node must have each of the
        ## indicated key-value pairs as labels.
        ## Read more here: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
        nodeSelector: { }
        ## Affinity settings for MinIO pods. Read more about affinity
        ## here: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity.
        affinity: { }
        ## Configure resource requests and limits for MinIO containers
        resources: { }
        ## Configure security context
        securityContext: { }
        ## Configure topology constraints
        topologySpreadConstraints: [ ]
    ## Mount path where PV will be mounted inside container(s).
    mountPath: /export
    ## Sub path inside Mount path where MinIO stores data.
    subPath: /data
    # pool secrets
    secrets:
      # create a kubernetes secret object with the accessKey and secretKey as defined here.
      enabled: true
      name: minio1-secret
      accessKey: minio
      secretKey: minio123
    # pool metrics to be read by Prometheus
    metrics:
      enabled: true
      port: 9000
    certificate:
      ## Use this field to provide one or more external CA certificates. This is used by MinIO
      ## to verify TLS connections with other applications:
      ## https://github.com/minio/minio/tree/master/docs/tls/kubernetes#2-create-kubernetes-secret
      externalCaCertSecret: { }
      ## Use this field to provide a list of Secrets with external certificates. This can be used to to configure
      ## TLS for MinIO Tenant pods. Create secrets as explained here:
      ## https://github.com/minio/minio/tree/master/docs/tls/kubernetes#2-create-kubernetes-secret
      externalCertSecret: { }
      ## Enable automatic Kubernetes based certificate generation and signing as explained in
      ## https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster
      requestAutoCert: false
      ## This field is used only when "requestAutoCert" is set to true. Use this field to set CommonName
      ## for the auto-generated certificate. Internal DNS name for the pod will be used if CommonName is
      ## not provided. DNS name format is *.minio.default.svc.cluster.local
      certConfig: { }
    ## Enable S3 specific features such as Bucket DNS which would allow `buckets` to be
    ## accessible as DNS entries of form `<bucketname>.minio.default.svc.cluster.local`
    s3:
      ## This feature is turned off by default
      bucketDNS: false
    ## List of bucket names to create during tenant provisioning
    buckets:
    - name: "tempo"
    - name: "loki"
```

Ideally, this would deploy the Tenant and create the buckets.

## Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->
Unfortunately, this does not succeed and instead the Tenant pods will appear to be running fine (based on readiness) but it seems they are not fine as `kubectl get tenants -n minio`  shows the state as stuck at `Provisioning default buckets`. This continues on for at least 8 minutes which is about as long as I've waited. After a minute or so, the Operator pods start to error and constantly keep restarting. The error shown in the Operator pods looks like this:

```
$ kubectl logs minio-operator-59fb779c44-xdnw5 -n minio
I0202 21:36:26.627415       1 main.go:71] Starting MinIO Operator
I0202 21:36:27.134091       1 main.go:150] caBundle on CRD updated
I0202 21:36:27.134663       1 main-controller.go:239] Setting up event handlers
I0202 21:36:27.134851       1 leaderelection.go:243] attempting to acquire leader lease minio/minio-operator-lock...
I0202 21:36:27.160652       1 leaderelection.go:253] successfully acquired lease minio/minio-operator-lock
I0202 21:36:27.160851       1 main-controller.go:450] minio-operator-59fb779c44-xdnw5: I've become the leader
I0202 21:36:27.161024       1 main-controller.go:365] Waiting for API to start
I0202 21:36:27.219783       1 main-controller.go:347] Starting HTTPS API server
I0202 21:36:27.220092       1 main-controller.go:369] Starting Tenant controller
I0202 21:36:27.220103       1 main-controller.go:372] Waiting for informer caches to sync
I0202 21:36:27.220110       1 main-controller.go:377] Starting workers
I0202 21:36:27.266626       1 status.go:120] Hit conflict issue, getting latest version of tenant
I0202 21:36:31.843819       1 status.go:120] Hit conflict issue, getting latest version of tenant
E0202 21:36:32.217250       1 runtime.go:78] Observed a panic: runtime.boundsError{x:0, y:0, signed:true, code:0x0} (runtime error: index out of range [0] with length 0)
goroutine 155 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x172d5c0, 0xc0004f63a8})
        k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:74 +0x85
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0009eaa40})
        k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:48 +0x75
panic({0x172d5c0, 0xc0004f63a8})
        runtime/panic.go:1038 +0x215
github.com/minio/operator/pkg/apis/minio.min.io/v2.(*Tenant).NewMinIOUserForAddress(0xc000d5a700, {0x0, 0x0}, {0x0, 0x0, 0x0}, {0xc000cb0900, 0x23a, 0x23b})
        github.com/minio/operator/pkg/apis/minio.min.io/v2/helper.go:715 +0x431
github.com/minio/operator/pkg/apis/minio.min.io/v2.(*Tenant).NewMinIOUser(...)
        github.com/minio/operator/pkg/apis/minio.min.io/v2/helper.go:703
github.com/minio/operator/pkg/controller/cluster.(*Controller).createBuckets(0xc0004c6360, {0x1a502d8, 0xc00009e000}, 0xc000d5a700, 0x0)
        github.com/minio/operator/pkg/controller/cluster/operator.go:328 +0x1fd
github.com/minio/operator/pkg/controller/cluster.(*Controller).syncHandler(0xc0004c6360, {0xc0009afb30, 0xc})
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:1278 +0x4a8a
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem.func1({0x15b88a0, 0xc0009eaa40})
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:543 +0x245
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem(0xc0004c6360)
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:555 +0x62
github.com/minio/operator/pkg/controller/cluster.(*Controller).runWorker(0x0)
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:495 +0x70
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7ff3dced4da0)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000628990, {0x1a237e0, 0xc000791d10}, 0x1, 0xc00008b500)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0, 0x3b9aca00, 0x0, 0x0, 0xc00075cca0)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0, 0x0, 0x0)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:90 +0x25
created by github.com/minio/operator/pkg/controller/cluster.(*Controller).Start.func1
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:380 +0x2dd
E0202 21:36:32.217353       1 runtime.go:78] Observed a panic: runtime.boundsError{x:0, y:0, signed:true, code:0x0} (runtime error: index out of range [0] with length 0)
goroutine 155 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x172d5c0, 0xc0004f63a8})
        k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:74 +0x85
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x4dcc74})
        k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:48 +0x75
panic({0x172d5c0, 0xc0004f63a8})
        runtime/panic.go:1038 +0x215
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0009eaa40})
        k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:55 +0xd8
panic({0x172d5c0, 0xc0004f63a8})
        runtime/panic.go:1038 +0x215
github.com/minio/operator/pkg/apis/minio.min.io/v2.(*Tenant).NewMinIOUserForAddress(0xc000d5a700, {0x0, 0x0}, {0x0, 0x0, 0x0}, {0xc000cb0900, 0x23a, 0x23b})
        github.com/minio/operator/pkg/apis/minio.min.io/v2/helper.go:715 +0x431
github.com/minio/operator/pkg/apis/minio.min.io/v2.(*Tenant).NewMinIOUser(...)
        github.com/minio/operator/pkg/apis/minio.min.io/v2/helper.go:703
github.com/minio/operator/pkg/controller/cluster.(*Controller).createBuckets(0xc0004c6360, {0x1a502d8, 0xc00009e000}, 0xc000d5a700, 0x0)
        github.com/minio/operator/pkg/controller/cluster/operator.go:328 +0x1fd
github.com/minio/operator/pkg/controller/cluster.(*Controller).syncHandler(0xc0004c6360, {0xc0009afb30, 0xc})
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:1278 +0x4a8a
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem.func1({0x15b88a0, 0xc0009eaa40})
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:543 +0x245
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem(0xc0004c6360)
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:555 +0x62
github.com/minio/operator/pkg/controller/cluster.(*Controller).runWorker(0x0)
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:495 +0x70
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7ff3dced4da0)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000628990, {0x1a237e0, 0xc000791d10}, 0x1, 0xc00008b500)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0, 0x3b9aca00, 0x0, 0x0, 0xc00075cca0)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0, 0x0, 0x0)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:90 +0x25
created by github.com/minio/operator/pkg/controller/cluster.(*Controller).Start.func1
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:380 +0x2dd
panic: runtime error: index out of range [0] with length 0 [recovered]
        panic: runtime error: index out of range [0] with length 0 [recovered]
        panic: runtime error: index out of range [0] with length 0

goroutine 155 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x4dcc74})
        k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:55 +0xd8
panic({0x172d5c0, 0xc0004f63a8})
        runtime/panic.go:1038 +0x215
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0009eaa40})
        k8s.io/apimachinery@v0.20.2/pkg/util/runtime/runtime.go:55 +0xd8
panic({0x172d5c0, 0xc0004f63a8})
        runtime/panic.go:1038 +0x215
github.com/minio/operator/pkg/apis/minio.min.io/v2.(*Tenant).NewMinIOUserForAddress(0xc000d5a700, {0x0, 0x0}, {0x0, 0x0, 0x0}, {0xc000cb0900, 0x23a, 0x23b})
        github.com/minio/operator/pkg/apis/minio.min.io/v2/helper.go:715 +0x431
github.com/minio/operator/pkg/apis/minio.min.io/v2.(*Tenant).NewMinIOUser(...)
        github.com/minio/operator/pkg/apis/minio.min.io/v2/helper.go:703
github.com/minio/operator/pkg/controller/cluster.(*Controller).createBuckets(0xc0004c6360, {0x1a502d8, 0xc00009e000}, 0xc000d5a700, 0x0)
        github.com/minio/operator/pkg/controller/cluster/operator.go:328 +0x1fd
github.com/minio/operator/pkg/controller/cluster.(*Controller).syncHandler(0xc0004c6360, {0xc0009afb30, 0xc})
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:1278 +0x4a8a
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem.func1({0x15b88a0, 0xc0009eaa40})
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:543 +0x245
github.com/minio/operator/pkg/controller/cluster.(*Controller).processNextWorkItem(0xc0004c6360)
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:555 +0x62
github.com/minio/operator/pkg/controller/cluster.(*Controller).runWorker(0x0)
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:495 +0x70
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7ff3dced4da0)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000628990, {0x1a237e0, 0xc000791d10}, 0x1, 0xc00008b500)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0, 0x3b9aca00, 0x0, 0x0, 0xc00075cca0)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0, 0x0, 0x0)
        k8s.io/apimachinery@v0.20.2/pkg/util/wait/wait.go:90 +0x25
created by github.com/minio/operator/pkg/controller/cluster.(*Controller).Start.func1
        github.com/minio/operator/pkg/controller/cluster/main-controller.go:380 +0x2dd
```

The logs for Tenant pod look like this:
```
$ kubectl logs minio1-ss-0-0 -n minio

 You are running an older version of MinIO released 3 weeks ago
 Update:
Run `mc admin update`



API: SYSTEM()
Time: 21:35:30 UTC 02/02/2022
Error: Marking http://minio1-ss-0-2.minio1-hl.minio.svc.cluster.local:9000/minio/storage/export1/data/v43 temporary offline; caused by Post "http://minio1-ss-0-2.minio1-hl.minio.svc.cluster.local:9000/minio/storage/export1/data/v43/readall?disk-id=&file-path=format.json&volume=.minio.sys": lookup minio1-ss-0-2.minio1-hl.minio.svc.cluster.local on 10.43.0.10:53: no such host (*fmt.wrapError)
       6: internal/rest/client.go:149:rest.(*Client).Call()
       5: cmd/storage-rest-client.go:152:cmd.(*storageRESTClient).call()
       4: cmd/storage-rest-client.go:520:cmd.(*storageRESTClient).ReadAll()
       3: cmd/format-erasure.go:406:cmd.loadFormatErasure()
       2: cmd/format-erasure.go:326:cmd.loadFormatErasureAll.func1()
       1: internal/sync/errgroup/errgroup.go:123:errgroup.(*Group).Go.func1()
```

Now if I comment out the buckets in my values.yaml and re-deploy the Tenant, everything works fine. I am able to tunnel into the console UI and add the buckets I need myself. However, as part of a scripted deployment, I want the buckets to be defined for me automatically.

## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
See above.

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->
I cannot seem to create buckets during Tenant deployment.

## Regression
<!-- Is this issue a regression? (Yes / No) -->
<!-- If Yes, optionally please include minio-operator version caused the regression, if you have these details. -->

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version used (`minio-operator`): 4.4.5
* Environment name and version (e.g. kubernetes v1.17.2): Kubernetes (K3s): v1.22.5+k3s1
* Server type and version:
* Operating System and version (`uname -a`): Linux k8s-poc-mgmt01 5.4.0-97-generic #110-Ubuntu SMP Thu Jan 13 18:22:13 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
* Link to your deployment file: See above

--- a/pkg/apis/minio.min.io/v2/helper.go
+++ b/pkg/apis/minio.min.io/v2/helper.go
@@ -648,12 +648,12 @@ func (t *Tenant) MinIOHealthCheck() bool {
 }
 
 // NewMinIOAdmin initializes a new madmin.Client for operator interaction
-func (t *Tenant) NewMinIOAdmin(minioSecret map[string][]byte) (*madmin.AdminClient, error) {
-	return t.NewMinIOAdminForAddress("", minioSecret)
+func (t *Tenant) NewMinIOAdmin(minioSecret map[string][]byte, caContent []byte) (*madmin.AdminClient, error) {
+	return t.NewMinIOAdminForAddress("", minioSecret, caContent)
 }
 
 // NewMinIOAdminForAddress initializes a new madmin.Client for operator interaction
-func (t *Tenant) NewMinIOAdminForAddress(address string, minioSecret map[string][]byte) (*madmin.AdminClient, error) {
+func (t *Tenant) NewMinIOAdminForAddress(address string, minioSecret map[string][]byte, caContent []byte) (*madmin.AdminClient, error) {
 	host, accessKey, secretKey, err := t.getMinIOTenantDetails(address, minioSecret)
 	if err != nil {
 		return nil, err
@@ -669,10 +669,32 @@ func (t *Tenant) NewMinIOAdminForAddress(address string, minioSecret map[string]
 		return nil, err
 	}
 
-	if opts.Secure {
-		// FIXME: add trusted CA
-		madmClnt = setUpInsecureTLS(madmClnt)
+	tr := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   2 * time.Second,
+			KeepAlive: 10 * time.Second,
+		}).DialContext,
+		ResponseHeaderTimeout: 5 * time.Second,
+		TLSHandshakeTimeout:   5 * time.Second,
+		ExpectContinueTimeout: 5 * time.Second,
+		// Go net/http automatically unzip if content-type is
+		// gzip disable this feature, as we are always interested
+		// in raw stream.
+		DisableCompression: true,
 	}
+	if opts.Secure {
+		rootCAs := mustGetSystemCertPool()
+		rootCAs.AppendCertsFromPEM(caContent)
+		tr.TLSClientConfig = &tls.Config{
+			// Can't use SSLv3 because of POODLE and BEAST
+			// Can't use TLSv1.0 because of POODLE and BEAST using CBC cipher
+			// Can't use TLSv1.1 because of RC4 cipher usage
+			MinVersion: tls.VersionTLS12,
+			RootCAs:    rootCAs,
+		}
+	}
+	madmClnt.SetCustomTransport(tr)
 
 	return madmClnt, nil
 }
@@ -712,36 +734,62 @@ func (t *Tenant) NewMinIOUserForAddress(address string, userCredentialSecrets []
 			return nil, errors.New("MinIO server host is empty")
 		}
 	}
-	consoleAccessKey, ok := userCredentialSecrets[0].Data["CONSOLE_ACCESS_KEY"]
-	if !ok {
-		return nil, errors.New("CONSOLE_ACCESS_KEY not provided")
-	}
-	// remove spaces and line breaks from access key
-	userAccessKey := strings.TrimSpace(string(consoleAccessKey))
-	consoleSecretKey, ok := userCredentialSecrets[0].Data["CONSOLE_SECRET_KEY"]
-	// remove spaces and line breaks from secret key
-	userSecretKey := strings.TrimSpace(string(consoleSecretKey))
-	if !ok {
-		return nil, errors.New("CONSOLE_SECRET_KEY not provided")
+
+	tr := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   2 * time.Second,
+			KeepAlive: 10 * time.Second,
+		}).DialContext,
+		ResponseHeaderTimeout: 5 * time.Second,
+		TLSHandshakeTimeout:   5 * time.Second,
+		ExpectContinueTimeout: 5 * time.Second,
+		// Go net/http automatically unzip if content-type is
+		// gzip disable this feature, as we are always interested
+		// in raw stream.
+		DisableCompression: true,
 	}
 
 	rootCAs := mustGetSystemCertPool()
 	rootCAs.AppendCertsFromPEM(caContent)
-	tr := &http.Transport{
-		TLSClientConfig: &tls.Config{},
-	}
-	tr.TLSClientConfig.RootCAs = rootCAs
-	opts := &minio.Options{
-		Transport: tr,
-		Secure:    t.TLS(),
-		Creds:     credentials.NewStaticV4(userAccessKey, userSecretKey, ""),
+
+	for _, cred := range userCredentialSecrets {
+		consoleAccessKey, ok := cred.Data["CONSOLE_ACCESS_KEY"]
+		if !ok {
+			return nil, errors.New("CONSOLE_ACCESS_KEY not provided")
+		}
+		// remove spaces and line breaks from access key
+		userAccessKey := strings.TrimSpace(string(consoleAccessKey))
+		consoleSecretKey, ok := cred.Data["CONSOLE_SECRET_KEY"]
+		// remove spaces and line breaks from secret key
+		userSecretKey := strings.TrimSpace(string(consoleSecretKey))
+		if !ok {
+			return nil, errors.New("CONSOLE_SECRET_KEY not provided")
+		}
+
+		tr.TLSClientConfig = &tls.Config{
+			// Can't use SSLv3 because of POODLE and BEAST
+			// Can't use TLSv1.0 because of POODLE and BEAST using CBC cipher
+			// Can't use TLSv1.1 because of RC4 cipher usage
+			MinVersion: tls.VersionTLS12,
+			RootCAs:    rootCAs,
+		}
+
+		opts := &minio.Options{
+			Transport: tr,
+			Secure:    t.TLS(),
+			Creds:     credentials.NewStaticV4(userAccessKey, userSecretKey, ""),
+		}
+
+		minioClnt, err := minio.New(host, opts)
+		if err != nil {
+			return nil, err
+		}
+
+		return minioClnt, nil
 	}
 
-	minioClnt, err := minio.New(host, opts)
-	if err != nil {
-		return nil, err
-	}
-	return minioClnt, nil
+	return nil, errors.New("no user credentials specified to initialize")
 }
 
 // mustGetSystemCertPool - return system CAs or empty pool in case of error (or windows)
@@ -891,30 +939,6 @@ func (t *Tenant) Validate() error {
 	}
 
 	return nil
-}
-
-// Set up admin client to use self certificates
-func setUpInsecureTLS(api *madmin.AdminClient) *madmin.AdminClient {
-	// Keep TLS config.
-	tlsConfig := &tls.Config{
-		// Can't use SSLv3 because of POODLE and BEAST
-		// Can't use TLSv1.0 because of POODLE and BEAST using CBC cipher
-		// Can't use TLSv1.1 because of RC4 cipher usage
-		MinVersion:         tls.VersionTLS12,
-		InsecureSkipVerify: true,
-	}
-
-	var transport http.RoundTripper = &http.Transport{
-		DialContext: (&net.Dialer{
-			Timeout:   10 * time.Second,
-			KeepAlive: 15 * time.Second,
-		}).DialContext,
-		TLSClientConfig: tlsConfig,
-	}
-
-	// Set custom transport.
-	api.SetCustomTransport(transport)
-	return api
 }
 
 // OwnerRef returns the OwnerReference to be added to all resources created by Tenant

--- a/pkg/controller/cluster/main-controller.go
+++ b/pkg/controller/cluster/main-controller.go
@@ -710,14 +710,27 @@ func (c *Controller) syncHandler(key string) error {
 		}
 	}
 
+	var caContent []byte
+	operatorCATLSCert, err := c.kubeClientSet.CoreV1().Secrets(miniov2.GetNSFromFile()).Get(ctx, "operator-ca-tls", metav1.GetOptions{})
+	// if custom ca.crt is not present in kubernetes secrets use the one stored in the pod
+	if err != nil {
+		caContent = miniov2.GetPodCAFromFile()
+	} else {
+		if val, ok := operatorCATLSCert.Data["ca.crt"]; ok {
+			caContent = val
+		}
+	}
+
 	tenantConfiguration, err := c.getTenantCredentials(ctx, tenant)
 	if err != nil {
 		return err
 	}
-	adminClnt, err := tenant.NewMinIOAdmin(tenantConfiguration)
+
+	adminClnt, err := tenant.NewMinIOAdmin(tenantConfiguration, caContent)
 	if err != nil {
 		return err
 	}
+
 	// For each pool check if there is a stateful set
 	var totalReplicas int32
 	var images []string
@@ -879,7 +892,7 @@ func (c *Controller) syncHandler(key string) error {
 					break
 				}
 				podAddress := fmt.Sprintf("%s:9000", tenant.MinIOHLPodHostname(ssPod.Name))
-				podAdminClnt, err := tenant.NewMinIOAdminForAddress(podAddress, tenantConfiguration)
+				podAdminClnt, err := tenant.NewMinIOAdminForAddress(podAddress, tenantConfiguration, caContent)
 				if err != nil {
 					return err
 				}
@@ -911,7 +924,7 @@ func (c *Controller) syncHandler(key string) error {
 							break
 						}
 						livePodAddress := fmt.Sprintf("%s:9000", tenant.MinIOHLPodHostname(livePod.Name))
-						livePodAdminClnt, err := tenant.NewMinIOAdminForAddress(livePodAddress, tenantConfiguration)
+						livePodAdminClnt, err := tenant.NewMinIOAdminForAddress(livePodAddress, tenantConfiguration, caContent)
 						if err != nil {
 							return err
 						}
@@ -1264,7 +1277,7 @@ func (c *Controller) syncHandler(key string) error {
 	}
 
 	// Ensure we are only provisioning users one time
-	if !tenant.Status.ProvisionedUsers {
+	if !tenant.Status.ProvisionedUsers && len(tenant.Spec.Users) > 0 {
 		if err := c.createUsers(ctx, tenant, tenantConfiguration); err != nil {
 			klog.V(2).Infof("Unable to create MinIO users: %v", err)
 			c.RegisterEvent(ctx, tenant, corev1.EventTypeWarning, "UsersCreatedFailed", fmt.Sprintf("Users creation failed: %s", err))
@@ -1274,7 +1287,7 @@ func (c *Controller) syncHandler(key string) error {
 	}
 
 	// Ensure we are only creating the bucket once using the console user
-	if !tenant.Status.ProvisionedBuckets && tenant.Status.ProvisionedUsers {
+	if !tenant.Status.ProvisionedBuckets && len(tenant.Spec.Buckets) > 0 {
 		if err := c.createBuckets(ctx, tenant, tenantConfiguration); err != nil {
 			klog.V(2).Infof("Unable to create MinIO buckets: %v", err)
 			c.RegisterEvent(ctx, tenant, corev1.EventTypeWarning, "BucketsCreatedFailed", fmt.Sprintf("Buckets creation failed: %s", err))

--- a/pkg/controller/cluster/monitoring.go
+++ b/pkg/controller/cluster/monitoring.go
@@ -115,7 +115,19 @@ func (c *Controller) updateHealthStatusForTenant(tenant *miniov2.Tenant) error {
 	if err != nil {
 		return err
 	}
-	adminClnt, err := tenant.NewMinIOAdmin(tenantConfiguration)
+
+	var caContent []byte
+	operatorCATLSCert, err := c.kubeClientSet.CoreV1().Secrets(miniov2.GetNSFromFile()).Get(context.Background(), "operator-ca-tls", metav1.GetOptions{})
+	// if custom ca.crt is not present in kubernetes secrets use the one stored in the pod
+	if err != nil {
+		caContent = miniov2.GetPodCAFromFile()
+	} else {
+		if val, ok := operatorCATLSCert.Data["ca.crt"]; ok {
+			caContent = val
+		}
+	}
+
+	adminClnt, err := tenant.NewMinIOAdmin(tenantConfiguration, caContent)
 	if err != nil {
 		// show the error and continue
 		klog.Infof("'%s/%s': %v", tenant.Namespace, tenant.Name, err)

--- a/pkg/controller/cluster/operator.go
+++ b/pkg/controller/cluster/operator.go
@@ -278,67 +278,92 @@ func (c *Controller) fetchUserCredentials(ctx context.Context, tenant *miniov2.T
 	return userCredentials
 }
 
-func (c *Controller) createUsers(ctx context.Context, tenant *miniov2.Tenant, tenantConfiguration map[string][]byte) error {
-	userCredentials := c.fetchUserCredentials(ctx, tenant)
+func (c *Controller) createUsers(ctx context.Context, tenant *miniov2.Tenant, tenantConfiguration map[string][]byte) (err error) {
+	defer func() {
+		if err == nil {
+			if _, err = c.updateProvisionedUsersStatus(ctx, tenant, true); err != nil {
+				klog.V(2).Infof(err.Error())
+			}
+		}
+	}()
 
-	if _, err := c.updateTenantStatus(ctx, tenant, StatusProvisioningInitialUsers, 0); err != nil {
+	userCredentials := c.fetchUserCredentials(ctx, tenant)
+	if len(userCredentials) == 0 {
+		return nil
+	}
+
+	if _, err = c.updateTenantStatus(ctx, tenant, StatusProvisioningInitialUsers, 0); err != nil {
 		return err
 	}
 
+	var caContent []byte
+	operatorCATLSCert, err := c.kubeClientSet.CoreV1().Secrets(miniov2.GetNSFromFile()).Get(ctx, "operator-ca-tls", metav1.GetOptions{})
+	// if custom ca.crt is not present in kubernetes secrets use the one stored in the pod
+	if err != nil {
+		caContent = miniov2.GetPodCAFromFile()
+	} else {
+		if val, ok := operatorCATLSCert.Data["ca.crt"]; ok {
+			caContent = val
+		}
+	}
+
 	// get a new admin client
-	adminClnt, err := tenant.NewMinIOAdmin(tenantConfiguration)
+	adminClnt, err := tenant.NewMinIOAdmin(tenantConfiguration, caContent)
 	if err != nil {
 		// show the error and continue
-		klog.Errorf("Error instantiating madmin: %v", err.Error())
+		klog.Errorf("Error instantiating madmin: %v", err)
+		return err
 	}
 
 	// configuration that means MinIO is running with LDAP enabled
 	// and we need to skip the console user creation
-	if err := tenant.CreateUsers(adminClnt, userCredentials); err != nil {
+	if err = tenant.CreateUsers(adminClnt, userCredentials); err != nil {
 		klog.V(2).Infof("Unable to create MinIO users: %v", err)
+	}
+
+	return err
+}
+
+func (c *Controller) createBuckets(ctx context.Context, tenant *miniov2.Tenant, tenantConfiguration map[string][]byte) (err error) {
+	defer func() {
+		if err == nil {
+			if _, err = c.updateProvisionedBucketStatus(ctx, tenant, true); err != nil {
+				klog.V(2).Infof(err.Error())
+			}
+		}
+	}()
+
+	if tenant.IsLDAPEnabled() {
+		return nil
+	}
+
+	if _, err := c.updateTenantStatus(ctx, tenant, StatusProvisioningDefaultBuckets, 0); err != nil {
 		return err
 	}
 
-	if _, err = c.updateProvisionedUsersStatus(ctx, tenant, true); err != nil {
-		klog.V(2).Infof(err.Error())
-	}
-
-	return nil
-}
-
-func (c *Controller) createBuckets(ctx context.Context, tenant *miniov2.Tenant, tenantConfiguration map[string][]byte) error {
-	// Skip default bucket creation for ldap setup
-	if !tenant.IsLDAPEnabled() && len(tenant.Spec.Buckets) > 0 {
-		if _, err := c.updateTenantStatus(ctx, tenant, StatusProvisioningDefaultBuckets, 0); err != nil {
-			return err
-		}
-		userCredentials := c.fetchUserCredentials(ctx, tenant)
-		var caContent []byte
-		operatorCATLSCert, err := c.kubeClientSet.CoreV1().Secrets(miniov2.GetNSFromFile()).Get(ctx, "operator-ca-tls", metav1.GetOptions{})
-		// if custom ca.crt is not present in kubernetes secrets use the one stored in the pod
-		if err != nil {
-			caContent = miniov2.GetPodCAFromFile()
-		} else {
-			if val, ok := operatorCATLSCert.Data["ca.crt"]; ok {
-				caContent = val
-			}
-		}
-
-		// Create bucket using the console user
-		minioClnt, err := tenant.NewMinIOUser(userCredentials, caContent)
-		if err != nil {
-			// show the error and continue
-			klog.Errorf("Error instantiating minio Client: %v", err.Error())
-		}
-
-		if err := tenant.CreateBuckets(minioClnt, tenant.Spec.Buckets...); err != nil {
-			klog.V(2).Infof("Unable to create MinIO buckets: %v", err)
-			return err
+	userCredentials := c.fetchUserCredentials(ctx, tenant)
+	var caContent []byte
+	operatorCATLSCert, err := c.kubeClientSet.CoreV1().Secrets(miniov2.GetNSFromFile()).Get(ctx, "operator-ca-tls", metav1.GetOptions{})
+	// if custom ca.crt is not present in kubernetes secrets use the one stored in the pod
+	if err != nil {
+		caContent = miniov2.GetPodCAFromFile()
+	} else {
+		if val, ok := operatorCATLSCert.Data["ca.crt"]; ok {
+			caContent = val
 		}
 	}
 
-	if _, err := c.updateProvisionedBucketStatus(ctx, tenant, true); err != nil {
-		klog.V(2).Infof(err.Error())
+	// Create bucket using the console user
+	minioClnt, err := tenant.NewMinIOUser(userCredentials, caContent)
+	if err != nil {
+		// show the error and continue
+		klog.Errorf("Error instantiating minio Client: %v, no buckets will be provisioned please verify if you have 'spec.users' set", err)
+		return err
+	}
+
+	if err := tenant.CreateBuckets(minioClnt, tenant.Spec.Buckets...); err != nil {
+		klog.V(2).Infof("Unable to create MinIO buckets: %v", err)
+		return err
 	}
 
 	return nil

